rocketpy/rocket/rocket.py

```diff
@@ -1916,38 +1925,78 @@ def to_dict(self, include_outputs=False):
             "sensors": self.sensors,
         }
 
-        if include_outputs:
+        if kwargs.get("include_outputs", False):
+            thrust_to_weight = self.thrust_to_weight
+            cp_position = self.cp_position
+            stability_margin = self.stability_margin
+            center_of_mass = self.center_of_mass
+            motor_center_of_mass_position = self.motor_center_of_mass_position
+            reduced_mass = self.reduced_mass
+            total_mass = self.total_mass
+            total_mass_flow_rate = self.total_mass_flow_rate
+            center_of_propellant_position = self.center_of_propellant_position
+
+            if discretize:
```

Comment:
I admit this part of the code is not exactly pretty, but it was the way I found that better guarantees the necessary variables will be discrete.